site/app/templates/notebook/Notebook.twig

```diff
@@ -262,6 +271,43 @@
 </div>
 
 <script>
+    //for displaying popup submit button
+    {% if is_timed and allowed_minutes > 25 %}
+        //interval is 20 percent of overall time allowed or 10 minutes, whichever is greater
+        let interval = Math.floor(((({{ allowed_minutes }}) / 5) > 10 ? ({{ allowed_minutes }}) / 5 : 10));
+        let loadTime = Date.now();
```

Comment:
Maybe this should be set to curTime as that is the official time from server while Date.now() is client time? You use curTime later on and Date.now() which could be differing values. Of course you would expect them to be different in order to calculate how long it has been but these could differ by pretty large amounts if the client set a different time. You could even get a negative time.

Reply:
when the page first loads curTime isn't set yet so I made it use Date.Now() once and then every time after that it uses curTime

Reply:
If you keep it this way, it is possible that there is already a x minute gap between Date.now() and curTime. It is possible for you to not start this until curTime is set by integrating the initial function call into submission-page.js. In probably 99/100 cases this won't be a problem. Timezone would not affect this but if someone's time isn't synced then there could be some issues.